Set up GitHub Actions

msasikanth opened the pull request:
Add 2 GH Actions

- CI Checks
- Publish library to Maven

Since there are no tests, at the moment CI checks only check for code formatting.

For publishing library GH Action, you do need to add these env variables in settings.
```
ORG_GRADLE_PROJECT_mavenCentralUsername=username
ORG_GRADLE_PROJECT_mavenCentralPassword=the_password

# see below for how to obtain this
ORG_GRADLE_PROJECT_signingInMemoryKey=exported_ascii_armored_key
# Optional
ORG_GRADLE_PROJECT_signingInMemoryKeyId=12345678
# If key was created with a password.
ORG_GRADLE_PROJECT_signingInMemoryKeyPassword=some_password
```
[more info](https://vanniktech.github.io/gradle-maven-publish-plugin/central/#secrets)

I was thinking of providing the version number as an input parameter as input to the publishing workflow. But wasn't sure if that's required or not, so skipped it for now.

msasikanth committed:
Add GitHub Action to run CI checks on main branch

msasikanth committed:
Add GitHub Action to publish the library to maven

msasikanth commented:
We can close #1 and create a new issue for creating tests for the library, WDYT?

ricknout commented:
> We can close #1 and create a new issue for creating tests for the library, WDYT?

Sounds good to me 👍🏼

msasikanth committed:
Publish library to Maven on push to the main branch

If the version has `-SNAPSHOT` it will publish to snapshot releases or else if it has a standard version, it will release that artifact.

msasikanth commented on .github/workflows/publish.yml:
```diff
@@ -0,0 +1,20 @@
+name: Release Library
+
+on:
+  workflow_dispatch:
```
Done @ricknout. This action should now make releases when a push to the main branch happens. We most likely don't need workflow_dispatch, but leaving it there for now manual trigger.

ricknout commented on .github/workflows/publish.yml:
```diff
@@ -0,0 +1,22 @@
+name: Release Library
```
nit: Publish Library

ricknout reviewed (approved):
Thanks for this! 🙏🏼 I'll add the env variables, branch protection check, etc once merged

msasikanth committed:
Address PR feedback
